### PR TITLE
Handling exceptions: providing invalid arguments for client API

### PR DESCRIPTION
**Pull request by AlexandrZagorskiy**

### CODE REVIEW

**Review by FalseR20 — APPROVED** [review]

**Comment by FallenChromium** [comment]
This PR was a result of having a hard-to-trace bug, and the problem being the wrong argument for the API method. There could be more cornercases like this though, wouldn't it be better to have a 100% method to check if the method was successful? The `status` field returned from the API will probably suffice, but there isn't a way to check for it right now. Alternatively we could throw an exception if status is 0, sounds a bit weird to me though.

**Review by FalseR20 — APPROVED** [review]

**Comment by AlexandrZagorskiy** [comment]
> This PR was a result of having a hard-to-trace bug, and the problem being the wrong argument for the API method. There could be more cornercases like this though, wouldn't it be better to have a 100% method to check if the method was successful? The `status` field returned from the API will probably suffice, but there isn't a way to check for it right now. Alternatively we could throw an exception if status is 0, sounds a bit weird to me though.

The _status_ field does not always represent that server returned error. Nikita mention it [here](https://github.com/ostis-ai/sc-machine/pull/137#discussion_r957826228). Returning status 0 without content means that command finished unsuccessfully, but in most cases, I think, this is a result of the wrong request from the user side, like a case with creating an empty list of the elements. You think that users should have access to the _status_ field, and it should be implemented? I think that raising errors is enough.

File: src/sc_client/client/_payload_factory.py [hunk 1]
```diff
@@ -29,6 +30,8 @@ def __call__(self, *args):
 
 class CreateElementsPayloadCreator(BasePayloadCreator):
     def __call__(self, constr: ScConstruction, *_):
+        if not isinstance(constr, ScConstruction):
```
Comment (NikitaZotov) [inline src/sc_client/client/_payload_factory.py:33]: Not all command cause exceptions

File: src/sc_client/client/_payload_factory.py [hunk 2]
```diff
@@ -29,6 +30,8 @@ def __call__(self, *args):
...
+            raise exceptions.InvalidTypeError("expected object types: ScConstruction")
```
Comment (NikitaZotov) [inline src/sc_client/client/_payload_factory.py:34]: Fix all errors catching in corresponding with logic in sc-server

**Review by NikitaZotov — CHANGES_REQUESTED** [review]

**Review by NikitaZotov — APPROVED** [review]

**Pull request merged**